### PR TITLE
[RHTAPBUGS-441]make changes to random chars

### PR DESCRIPTION
**Pull request by yangcao77**
### What does this PR do?:

Apply the changes that if `generateComponentName` has been set, do not check for component existence and directly generate component name with random chars appended. 

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/RHTAPBUGS-441

### PR acceptance criteria:

- [x] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: controllers/update.go [hunk 1]
```diff
@@ -499,7 +503,11 @@ func (r *ComponentDetectionQueryReconciler) updateComponentStub(req ctrl.Request
 			Revision:      componentDetectionQuery.Spec.GitSource.Revision,
 			DockerfileURL: link,
 		}
-		componentName := r.getComponentName(log, ctx, req.Namespace, gitSource)
+		generateComponentName := false
+		if componentDetectionQuery.Spec.GenerateComponentName {
+			generateComponentName = componentDetectionQuery.Spec.GenerateComponentName
+		}
```
Comment (maysunfaisal) [inline controllers/update.go:509]: cant this block be declared once for both devfile map and dockerfile map?

Comment (yangcao77) [reply controllers/update.go:509]: they are in different functions. updated the PR to move this block to `getComponentName` method.

**Review by maysunfaisal — APPROVED** [review]

**Comment by yangcao77** [comment]
@maysunfaisal  may i get another lgtm label? thx

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**